src/olympia/amo/tests/test_utils_.py

```diff
@@ -172,3 +175,19 @@ def bar(self):
     del foo.bar
     assert foo.bar == 'original value'
     assert callme.call_count == 1
+
+
+@pytest.mark.parametrize('lang', settings.AMO_LANGUAGES)
+def test_get_locale_from_lang(lang):
+    """Make sure all languages in settings.AMO_LANGUAGES can be resolved."""
+    locale = get_locale_from_lang(lang)
+
+    assert isinstance(locale, Locale)
+    assert locale.language == lang[:2]
```

Comment:
The thing is, `babel` expects the language code to be encoded as `en_US`, not `en-US` which is why we use `django.translation.to_locale` in `get_locale_from_lang` to convert it.

Imho `get_locale_from_lang` has always been broken and should have used `Locale.parse`, especially for `zh-CN` and `zh-TW` since those aren't the actual locale codes but shortcuts for `zh-Hans-CN` and only `Locale.parse` understands that. Well, in newer versions of babel at least, it appears older versions supported those as an argument to `Locale(...)`.

Does that answer your question? :laughing: